pupil-spa/src/app/storage.service.ts

```diff
@@ -3,40 +3,30 @@ import { Inject, Injectable } from '@angular/core';
 @Injectable()
 export class StorageService {
 
-  setItem(key, value): Promise<any> {
-    return new Promise((resolve, reject) => {
-      if (!key) {
-        const err = new Error('key is required');
-        reject(err);
-      } else {
-        localStorage.setItem(key, value);
-        resolve();
-      }
-    });
+  setItem(key, value): void {
+
+    if (!key) {
+      throw new Error('key is required');
+    } else {
```

Comment:
There is no need for the `else` branch - it can just read 

```
if (bailoutCondition) {
 throw ()
}
return expression
```

This is cleaner code I think.